tests/utils/spark_operator.go

```diff
@@ -1,135 +1,158 @@
 package utils
 
 import (
+	"errors"
+	"fmt"
 	log "github.com/sirupsen/logrus"
-	coreV1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"os"
 	"os/exec"
+	"path"
+	"regexp"
 	"strings"
+	"time"
 )
 
 type SparkOperatorInstallation struct {
+	Namespace    string
+	InstanceName string
+	Clients      *kubernetes.Clientset
+}
+
+type SparkInstance struct {
+	Name      string
 	Namespace string
-	Clients   *kubernetes.Clientset
+	Status    string
 }
 
-func InstallSparkOperator() *SparkOperatorInstallation {
+func InstallSparkOperator() (*SparkOperatorInstallation, error) {
 	return InstallSparkOperatorWithNamespace(DefaultNamespace)
 }
 
-func InstallSparkOperatorWithNamespace(namespace string) *SparkOperatorInstallation {
+func InstallSparkOperatorWithNamespace(namespace string) (*SparkOperatorInstallation, error) {
 	clientSet, err := GetK8sClientSet()
 	if err != nil {
 		log.Fatal(err)
 		panic(err)
 	}
 
-	err = installSparkOperatorWithHelm(namespace)
-	if err != nil {
-		log.Fatal(err)
-		panic(err)
-	}
-
 	spark := SparkOperatorInstallation{
 		Namespace: namespace,
 		Clients:   clientSet,
 	}
-	return &spark
-}
+	spark.CleanUp()
 
-func (spark *SparkOperatorInstallation) CleanUp() {
-	uninstallSparkOperatorWithHelm(spark.Namespace)
+	installScript := exec.Command("bash", path.Join(TestDir, "../scripts/install_operator.sh"))
```

Comment:
Hi @akirillov 
> So it would be great to have the following:
> 
> * migrate off from using `install_operator.sh` script and
>   

I also agree that once `kudo install <operater>` will be good enough to install fully working operator(with ability to enable/disable modes by a parameters specifying) it should be used in tests to spin up the operator. However, as for now, we have a couple of workarounds accumulated in `install_operator.sh`, `remove_operator.sh` like monitoring service, crds installation etc. And I think it was a bit early to migrate from it now. Because currently we are often make changes to the `install_operator.sh` and it will not be reflected to here. I mean we should maintain these workarounds in two places now. Just IMO.